unblob/handlers/filesystem/yaffs/utils.py

```diff
@@ -0,0 +1,574 @@
+                )
+                break
+
+        if config is None:
+            return self.bruteforce()
+
+        # Now to try to identify the spare data size...
+        # If not using the ECC layout, there are 2 extra bytes at the beginning of the
+        # spare data block. Ignore them.
+
+        ecc_offset = 0 if config.ecc else 2
+
+        # TODO: it works except for 512/16
+
+        # The spare data signature is built dynamically, as there are repeating data patterns
+        # that we can match on to find where the spare data ends. Take this hexdump for example:
+        #
+        # 00000800  00 10 00 00 01 01 00 00  00 00 00 00 ff ff ff ff  |................|
+        # 00000810  03 00 00 00 01 01 00 00  ff ff 62 61 72 00 00 00  |..........bar...|
+        # 00000820  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
+        #
+        # The spare data starts at offset 0x800 and is 16 bytes in size. The next page data then
+        # starts at offset 0x810. Not that the four bytes at 0x804 (in the spare data section) and
+        # the four bytes at 0x814 (in the next page data section) are identical. This is because
+        # the four bytes at offset 0x804 represent the object ID of the previous object, and the four
+        # bytes at offset 0x814 represent the parent object ID of the next object. Also, the
+        # four bytes in the page data are always followed by 0xFFFF, as those are the unused name
+        # checksum bytes.
+        #
+        # Thus, the signature for identifying the next page section (and hence, the end of the
+        # spare data section) becomes: [the 4 bytes starting at offset 0x804] + 0xFFFF
+        #
+        # Note that this requires at least one non-empty subdirectory; in practice, any Linux
+        # file system should meet this requirement, but one could create a file system that
+        # does not meet this requirement.
+
+        object_id_offset = 4
+        object_id_start = page_size + ecc_offset + object_id_offset
+        object_id_end = object_id_start + 4
+        spare_signature = self.file[object_id_start:object_id_end] + b"\xFF\xFF"
+
+        config.spare_size = (
+            self.file[object_id_end : object_id_end + page_size].find(spare_signature)
+            + object_id_offset
+            + ecc_offset
+        )
+
+        # Sanity check the spare size, make sure it looks legit
+        if config.spare_size not in VALID_SPARE_SIZES:
+            raise InvalidInputFormat(
+                "Auto-detection failed: Detected an unlikely spare size: %d"
+                % config.spare_size
+            )
+
+        return config
+
+    def insert_entry(self, entry: YAFFSEntry):
+        try:
+            self.file_entries.create_node(
+                entry.object_id,
+                entry.object_id,
+                data=entry,
+                parent=entry.parent_obj_id,
+            )
+        except NodeIDAbsentError:
```

Comment:
how could this happen, why ignore it?

Reply:
It can't happen, I removed that code. I also fixed my code around tree handling so that we no longer need to handle these errors. If they are raised, something is wrong and it's legit to raise an `InvalidInputFormat`.